src/backend/packages/functions/src/appointment.ts

```diff
@@ -0,0 +1,83 @@
+import { APIGatewayProxyHandler, Context } from 'aws-lambda'
+import { APIGatewayProxyEvent } from 'aws-lambda'
+
+function getPathWithoutParams(path: string, pathParameters: { [name: string]: string } | null) {
+  let result = path
+
+  if (pathParameters) {
+    for (const param in pathParameters) {
+      result = result.replace(pathParameters[param], `{${param}}`)
+    }
+  }
+
+  return result
+}
+
+export interface SSTAPIGatewayProxyEvent {
+  rawPath: string
+  pathParameters: { [name: string]: string } | null
+  requestContext: {
+    http: {
+      method: string
+    }
+  }
+  queryStringParameters: { [name: string]: string } | null // Add this line
+  body: string | null
+}
+
+type SSTAPIGatewayProxyHandler = (
+  event: SSTAPIGatewayProxyEvent,
+  context: Context,
+) => Promise<APIGatewayProxyEvent | { [key: string]: any }>
+
+export const handler: SSTAPIGatewayProxyHandler = async (event: SSTAPIGatewayProxyEvent) => {
+  const httpMethod = event.requestContext.http.method
+  const path = getPathWithoutParams(event.rawPath, event.pathParameters)
+  const queryParams = event.queryStringParameters
+  console.log(httpMethod)
+  console.log(path)
+  if (httpMethod === 'GET' && path === '/appointment/{id}') {
```

Comment:
handling GET and POST together is terrible.. is the solution to have more scaffolding with patch.ts and post.ts in /appointment folder?

Reply:
yes I would create separate handler functions for the different methods, and a separate lambda to handle each